### ISSUE TITLE
MethodError when using Float32 numbers with ROS3P solver.

### ISSUE THREAD

**Issue by IlyaOrson** [SciML/OrdinaryDiffEq.jl#1335]
MWE:
```julia
julia> using OrdinaryDiffEq

julia> init(ODEProblem((u,p,t)->u,(p,t0)->p[1],(p)->(0f0,p[2]),(2f0,1f0)), ROS3P())
ERROR: MethodError: no method matching OrdinaryDiffEq.ROS3PTableau(::Float32, ::Float32, ::Float32, ::Float32, ::Float64, ::Float64, ::Float64, ::Float64, ::Float64, ::Float64, ::Float64, ::Float64, ::Float32, ::Float32, ::Float32, ::Float32, ::Float32, ::Float32)
Closest candidates are:
  OrdinaryDiffEq.ROS3PTableau(::T, ::T, ::T, ::T, ::T, ::T, ::T, ::T, ::T, ::T, ::T, ::T, ::T2, ::T2, ::T2, ::T, ::T, ::T) where {T, T2} at C:\Users\ilyao\.julia\packages\OrdinaryDiffEq\RGFoW\src\tableaus\rosenbrock_tableaus.jl:24

(@v1.5) pkg> st OrdinaryDiffEq
Status `C:\Users\ilyao\.julia\environments\v1.5\Project.toml`
  [1dea7af3] OrdinaryDiffEq v5.49.1
```

If I convert to type T the fractions in https://github.com/SciML/OrdinaryDiffEq.jl/blob/195852bca58e13569b244fdfa5546298c8941bea/src/tableaus/rosenbrock_tableaus.jl#L51-L57 it works locally.

I would like to make the PR. Do I need to add test for this somewhere though?

**Comment by ChrisRackauckas** [comment]
Yeah that would be a good test. I don't know if we have a general test for Float32 compatibility, but it would be good to add. It might be good to expand https://github.com/SciML/OrdinaryDiffEq.jl/blob/master/test/interface/ode_numbertype_tests.jl to loop over number types and solvers, like https://github.com/SciML/OrdinaryDiffEq.jl/blob/master/test/interface/units_tests.jl

**Comment by ChrisRackauckas-Claude** [comment]
Fixed — the exact example from the report now initializes and solves, with no `ROS3PTableau` MethodError:

```
init:  SUCCESS, eltype(u0)=Float32, tspan=(0.0f0, 1.0f0)
solve: retcode=Success, u(end)=5.263968, typeof=Float32
```

One thing I chased down and cleared, in case it looks alarming to anyone re-testing this: ROS3P appears inaccurate on `u'=1.01u` (~12% error at tol=1e-8, with step count nearly flat in tolerance). That is **not** a bug and not Float32-specific — Float64 behaves identically, fixed-dt convergence is a clean order 3 (`orders=[3.09, 3.04, 3.02]`), and on nonlinear problems error tracks tolerance properly (tol 1e-4 -> 1e-10 gives err 1.7e-4 -> 3.2e-10). It is the embedded error estimator degenerating on a scalar *linear* problem, unrelated to this issue.

Verified on OrdinaryDiffEq v7.1.3 / Julia 1.11.9. Closing.
